### PR TITLE
Added new type of enemy called the Diver

### PR DESCRIPTION
**Pull request by dudghks**
Created a new type of enemy called the Diver
- Movement and Attack pattern
    - Divers remain at the top of the screen and move in one continuous direction until they either reach the end of the screen or bump into another diver
    - Divers will attack every 5-8 seconds
    - Attacks are "charged" for roughly 1 second. During this time, they will flash red and white
    - Divers attack by charge downwards in a straight line (they appear red during this). Contact with the player will deal damage.
    - They charge through the bottom of the screen and loop back to the top, ending the attack once they reach their original position
- TODO
    - Update sprite
    - Change count based on level/difficulty
    - Change charge speed based on difficulty

### CODE REVIEW

**Review by brewcoua — APPROVED** [review]
Looks good to me :ok_hand:

**Review by brewcoua — APPROVED** [review]
LGTM :ok_hand:

**Comment by dudghks** [comment]
Recent changes (cleared TODO list):
Update sprite
Change count based on level/difficulty
Change charge speed based on difficulty

File: src/entity/EnemyShipDiver.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        if (this.animationCooldown.checkFinished()) {
+            this.animationCooldown.reset();
+
+            switch (this.spriteType) {
+                case EnemyShipE1:
+                    this.spriteType = DrawManager.SpriteType.EnemyShipE2;
+                    break;
+                case EnemyShipE2:
+                    this.spriteType = DrawManager.SpriteType.EnemyShipE1;
```
Comment (brewcoua) [inline src/entity/EnemyShipDiver.java]: This looks good to me, but would it not be better to use the `EnemyShipF1` sprite instead of `EnemyShipE1` ?

Comment (dudghks) [reply src/entity/EnemyShipDiver.java]: That could work -- I chose E1 because it F1 was noted as an elite enemy in the DrawManager. I'll change it to F1

**Review by brewcoua — APPROVED** [review]
LGTM :ok_hand:

**Pull request merged**